virttest/utils_libvirt/libvirt_vmxml.py

```diff
@@ -21,3 +23,19 @@ def set_vm_attrs(vmxml, vm_attrs):
     vmxml.xmltreefile.write()
     vmxml.sync()
     return vmxml
+
+
+def remove_vm_devices_by_type(vm, device_type):
+    """
+    Remove devices of a given type.
+
+    :param vm_name: The vm object.
+    :param device_type: Type of devices should be removed.
+    """
+    vm_was_running = vm.is_alive()
+    vmxml = vm_xml.VMXML.new_from_dumpxml(vm.name)
+    vmxml.remove_all_device_by_type(device_type)
+    vmxml.sync()
+
+    if vm_was_running:
+        vm.start()
```

Comment:
Alive VM is not destroyed before removing a device, why is it started again

Reply:
vmxml.sync() will destroy the vm(actually, it's destroyed by virsh.remove_domain).